src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs

```diff
@@ -821,6 +814,28 @@ await Win32API.SendMessageAsync(connection, new ValueSet()
             }
         }
 
+        private ShellItem GetFirstFile(ShellItem shi)
+        {
+            if (!shi.IsFolder || shi.Attributes.HasFlag(ShellItemAttribute.Stream))
+            {
+                return shi;
+            }
+            using var shf = new ShellFolder(shi);
+            if (shf.FirstOrDefault(x => !x.IsFolder || x.Attributes.HasFlag(ShellItemAttribute.Stream)) is ShellItem item)
+            {
+                return item;
+            }
+            foreach (var shsfi in shf.Where(x => x.IsFolder && !x.Attributes.HasFlag(ShellItemAttribute.Stream)))
+            {
+                using var shsf = new ShellFolder(shsfi);
+                if (GetFirstFile(shsf) is ShellItem item2)
```

Comment:
Looks cool! Would it be possible to do this using a loop instead of recursion?

Reply:
Probably, I understand that there's a programming theorem that says "every recursion be converted into iteration" :) In this case I find recursion more intuitive, but can be changed if you think it's necessary.

Reply:
Ah, can be left as is :)